src/Storage/Service/AbstractUrlStorage.php

```diff
@@ -132,19 +161,32 @@ public function addChunk(string $hash, string $chunk): bool
 
     public function finalizeUpload(string $hash): bool
     {
+        if ($this->isReadOnly()) {
+            return false;
+        }
         $source = $this->getUploadPath($hash);
         $destination  = $this->getPath($hash);
         $this->initDirectory($destination);
         try {
             return \rename($source, $destination);
         } catch (\Throwable $e) {
-            //TODO: add log info or notice
+            $this->logger->info(sprintf('Rename %s to %s failed: %s', $source, $destination, $e->getMessage()));
         }
         try {
             return \copy($source, $destination);
         } catch (\Throwable $e) {
-            //TODO: add log info or notice
+            $this->logger->info(sprintf('Copy %s to %s failed: %s', $source, $destination, $e->getMessage()));
         }
         return false;
     }
+
+    public function isReadOnly(): bool
+    {
+        return $this->readOnly;
+    }
+
+    public function isToSkip(): bool
```

Comment:
This would become "shouldSkip()"

Reply:
Agree (and it's the reason of toSkip, isSkip was weird) naming issue 96993866c927751ec0fac5a9886e27fdc4d771b8